server/src/main/java/io/crate/replication/logical/engine/SubscriberEngine.java

```diff
@@ -67,12 +70,17 @@ static void validate(final Operation operation) {
     @Override
     protected IndexingStrategy indexingStrategyForOperation(Index index) throws IOException {
         validate(index);
-        return super.indexingStrategyForOperation(index);
+        return planIndexingAsNonPrimary(index);
```

Comment:
The pre-flight check caused assertion errors in the replication tests. Changed this similar to https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/ReplicationEngine.kt